src/main/java/in/projecteka/gateway/registry/RegistryRepository.java

```diff
@@ -0,0 +1,38 @@
+package in.projecteka.gateway.registry;
+
+import in.projecteka.gateway.common.DbOperationError;
+import in.projecteka.gateway.common.MappingRepository;
+import in.projecteka.gateway.registry.Model.CMServiceRequest;
+import io.vertx.pgclient.PgPool;
+import io.vertx.sqlclient.Tuple;
+import lombok.AllArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+
+@AllArgsConstructor
+public class RegistryRepository {
+    private static final Logger logger = LoggerFactory.getLogger(MappingRepository.class);
+    private static final String UPSERT_CM_ENTRY =
+            "INSERT INTO consent_manager (id, name, url, cm_id, suffix, active, blocklisted, license, license_authority)" +
+                    "VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9) ON CONFLICT (cm_id) DO" +
+                    "UPDATE SET name = $2, url = $3, active = $6, blocklisted = $7, date_modified = timezone('utc'::text, now())" +
+                    "WHERE consent_manager.cm_id = $4";
```

Comment:
This is not a preferred approach relying on the underlying database, if database changes (which is highly unlikely), still it's better read whether you have cm record, if yes update, otherwise insert it.

Reply:
And suffix is the main thing here, if you already have an entry with a suffix, then update, if not create it.

I'm not sure why have id in first place. You can check with @angshu

Reply:
done